pandas/tests/frame/test_join.py

```diff
@@ -97,12 +97,6 @@ def test_join_index(frame):
     with pytest.raises(ValueError, match='join method'):
         f.join(f2, how='foo')
 
-    # corner case - overlapping columns
```

Comment:
hmm, this used to error. I am not sure I like this change. cc @jorisvandenbossche

Reply:
yes, this removal is made because in last PR we found out pandas don't explicitly refuse overlapping columns, e.g.
```python
df1 = pd.DataFrame({"a": [1, 2], "b": [3, 4], "b_x": [5, 6]})
df2 = pd.DataFrame({"a": [1, 2], "b": [4, 5]})
df1.merge(df2, on="a")
```
the output will still contain two "b_x" columns. @jorisvandenbossche @simonjayhawkins